test/fail_if_no_progress.lean

```diff
@@ -1,12 +1,18 @@
 import Mathlib.Tactic.FailIfNoProgress
+import Mathlib.Data.List.Basic
 
```

Comment:
I'm not going to insist, but it seems the ratio of configuration options to tests is high! We could add more exhaustive tests, or perhaps just try next time to resist the temptation to add so many knobs that we can't even be bothered testing them. :-)